File: backend/src/main/java/com/stenico/ventas/model/Cheque.java
package com.stenico.ventas.model;

import java.math.BigDecimal;
import java.time.LocalDate;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "cheques")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Cheque {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Column(nullable = false)
    private String banco;

    @NotBlank
    @Column(nullable = false, length = 40)
    private String numero;

    private String titular;

    private String numeroCuenta;

    private LocalDate fechaEmision;

    private LocalDate fechaVencimiento;

    @NotNull
    @Column(nullable = false)
    private BigDecimal monto;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private EstadoCheque estado = EstadoCheque.PENDIENTE;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "venta_id")
    private Venta venta;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "cliente_id")
    private Cliente cliente;
}

File: frontend/src/app/models/cheque.ts
import { Venta } from './venta';
import { Cliente } from './cliente';

export type EstadoCheque = 'PENDIENTE' | 'COBRADO' | 'RECHAZADO' | 'CANCELADO';

export interface Cheque {
  id?: number;
  banco: string;
  numero: string;
  titular?: string;
  numeroCuenta?: string;
  fechaEmision?: string;
  fechaVencimiento?: string;
  monto: number;
  estado: EstadoCheque;
  venta?: Venta;
  cliente?: Cliente;
}

export interface ChequeRequest {
  banco: string;
  numero: string;
  titular?: string;
  numeroCuenta?: string;
  fechaEmision?: string;
  fechaVencimiento?: string;
  monto: number;
  ventaId?: number;
  clienteId?: number;
  estado?: EstadoCheque;
}
